android/src/main/java/com/lwansbrough/RCTCamera/RCTCameraModule.java

```diff
@@ -175,6 +197,211 @@ public String getName() {
         });
     }
 
+    private Throwable prepareMediaRecorder(ReadableMap options) {
+        CamcorderProfile cm = RCTCamera.getInstance().setCaptureVideoQuality(options.getInt("type"), options.getString("quality"));
+
+        // Attach callback to handle maxDuration (@see onInfo method in this file)
+        mMediaRecorder.setOnInfoListener(this);
+        mMediaRecorder.setCamera(mCamera);
+
+        mCamera.unlock();  // make available for mediarecorder
+
+        // Set AV sources
+        mMediaRecorder.setAudioSource(MediaRecorder.AudioSource.CAMCORDER);
+        mMediaRecorder.setVideoSource(MediaRecorder.VideoSource.CAMERA);
+
+        int mediaRecorderHintOrientation = (90 + ((720 - RCTCamera.getInstance().getActualDeviceOrientation() * 90))) % 360;
+        // adjust for differences in how devices display front facing http://www.theverge.com/2015/11/9/9696774/google-nexus-5x-upside-down-camera
+        if (RCT_CAMERA_TYPE_FRONT == options.getInt("type")) {
+            if ((RCT_CAMERA_ORIENTATION_PORTRAIT == RCTCamera.getInstance().getOrientation()) && (RCTCamera.getInstance().getAdjustedDeviceOrientation() == 90)) {
+                mediaRecorderHintOrientation += 180;
+            }
+        }
+
+        mMediaRecorder.setOrientationHint(mediaRecorderHintOrientation);
+
+        if (cm == null) {
+            return new RuntimeException("CamcorderProfile not found in prepareMediaRecorder.");
+        }
+
+        cm.fileFormat = MediaRecorder.OutputFormat.MPEG_4;
+        cm.videoCodec = MediaRecorder.VideoEncoder.MPEG_4_SP;
```

Comment:
Ok, I was able to fix my no-recording problem by commenting out this line (fix tested on Nexus 9 and Oneplus One). It appears my hardware doesn't have support for the `MPEG_4_SP` codec. Is this line absolutely necessary on some configurations, or should we just leave it out and let the system choose the default codec?